dashboard/routes/wallet/create.go

```diff
@@ -33,17 +36,73 @@ func WalletCreateRouter(router *gin.RouterGroup) {
 			log.WebLog.Error("wallet file exists.")
 			context.AbortWithError(http.StatusInternalServerError, errors.New("wallet file exists."))
 			return
-		} else {
-			_, err := vault.NewWallet(config.Parameters.WalletFile, []byte(data.Password), true)
+		}
+
+		if config.Parameters.WebGuiCreateWallet && !config.Parameters.AllowEmptyBeneficiaryAddress {
+			if data.BeneficiaryAddr == "" {
+				log.WebLog.Error("beneficiary address is empty.")
+				context.AbortWithError(http.StatusBadRequest, errors.New("beneficiary address is empty."))
+				return
+			}
+
+			_, err = ToScriptHash(data.BeneficiaryAddr)
```

Comment:
From L48 to L85 is the same as set beneficiary in BeneficiaryRouter, what about making a common function so both place can simply call it?